src/download/download.service.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import type { Response } from "express";
+import { FileEntity } from "src/file/file.entity";
+import { Repository } from "typeorm";
+import { BUCKET_DIRECTORY, MESSAGE, UPLOAD_TYPE } from "src/constants";
+
+@Injectable()
+export class DownloadService {
+  constructor(
+    @InjectRepository(FileEntity)
+    private readonly fileRepository: Repository<FileEntity>
+  ) {}
+
+  /**
+   * Function to download a file
+   * @param res Response of a request
+   * @param fileId unique fileId
+   */
+  async downloadFile(res: Response, fileId: string): Promise<void> {
+    try {
+      const fileObj = await this.fileRepository.findOne(fileId);
+
+      switch (fileObj?.uploadingStatus) {
+        case UPLOAD_TYPE.COMPLETED:
+          const path = `${BUCKET_DIRECTORY}/${fileObj.filenameOnBucket}`;
+          res.download(path);
```

Comment:
This will only work when file is open to everyone. Read file buffer and download the same on FE

Reply:
Can you please explain  meaning of  the file is open to everyone here?

- Read the file buffer and download the same on FE 
 ->  I think internally it is working the same. it is actually streaming a file.  
 
 Link: https://github.com/expressjs/express/blob/master/lib/response.js#L1042

Reply:
LGTM